src/validation.cpp

```diff
@@ -167,6 +167,24 @@ namespace {
     std::set<int> setDirtyFileInfo;
 } // anon namespace
 
+CCoinsViewMemPool::CCoinsViewMemPool(CCoinsView* baseIn, const CTxMemPool& mempoolIn) : CCoinsViewBacked(baseIn), mempool(mempoolIn) { }
+
+bool CCoinsViewMemPool::GetCoin(const COutPoint &outpoint, Coin &coin) const {
+    // If an entry in the mempool exists, always return that one, as it's guaranteed to never
+    // conflict with the underlying cache, and it cannot have pruned entries (as it contains full)
+    // transactions. First checking the underlying cache risks returning a pruned entry instead.
```

Comment:
Everything after ", and it cannot have pruned ..." can be removed. There's no such thing as pruned coins cache entries since per-txout db.

Reply:
Oh true. I do fix this in #20833... should I do it here instead?

Reply:
Up to you. I only mention it here because you're touching this code, but if you want to keep this move-only, then that's fine as well.